File: Problem-463-Islam_Perimeter/my_sol.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        
        int rows = grid.size();
        int cols = grid[0].size();
        int p = 0; 

        for(int i=0; i<rows; i++)
        {
            for(int j=0; j<cols; j++)
            {
                if(grid[i][j] == 1)
                {
                    //top
                    if( i==0 || grid[i-1][j]==0)
                    {
                        p++;
                    }
                    //buttom
                    if( i==rows-1 || grid[i+1][j] == 0)
                    {
                        p++;
                    }
                    //left
                    if( j==0 ||grid[i][j-1] == 0)
                    {
                        p++;
                    }
                    //right
                    if( j == cols-1 || grid[i][j+1] == 0)
                    {
                        p++;
                    }
                }

            }
        }
        return p;
    }
};

//time complexity: O(rows*cols)
//space complexity: O(1)


//This time my solution is the optimimal one .Yoo.

File: Problem-1456-Maximum Number of Vowels in a Substring of Given Length/my_sol.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        left = 0
        right = k-1
        n = len(s) 
        vowel = 0
        maximum = - 9999

        i = left
        while i <= right:
            if s[i] in ['a', 'e', 'i', 'o', 'u']:  
                vowel = vowel + 1
            i += 1

        maximum = max(maximum, vowel)

        left += 1
        right += 1

        while right < n:
            if s[left-1] in ['a', 'e', 'i', 'o', 'u']:  
                vowel -= 1
            if  s[right] in ['a', 'e', 'i', 'o', 'u']:  
                vowel += 1
            
            maximum = max(maximum, vowel)

            left += 1
            right += 1
        

        return maximum


#time complexity: O(n)
#space complexity: O(1)